lib/puppetserver/ca/list_action.rb

```diff
@@ -0,0 +1,136 @@
+require 'puppetserver/ca/utils'
+require 'puppetserver/utils/http_utilities'
+require 'puppetserver/utils/file_utilities'
+require 'puppetserver/ca/puppet_config'
+require 'optparse'
+require 'json'
+
+module Puppetserver
+  module Ca
+    class ListAction
+
+      include Puppetserver::Utils
+
+      SUMMARY = 'List all certificate requests'
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca list [--help|--version]
+  puppetserver ca list [--config]
+  puppetserver ca list [--all]
+
+Description:
+List outstanding certificates. If --all is specified, signed certificates will be listed as well.
+
+Options:
+      BANNER
+
+      BODY = JSON.dump({desired_state: 'signed'})
+
+      def initialize(logger)
+        @logger = logger
+      end
+
+      def self.parser(parsed = {})
+        OptionParser.new do |opts|
+          opts.banner = BANNER
+          opts.on('--config PUPPET.CONF', 'Custom path to Puppet\'s config file') do |conf|
+            parsed['config'] = conf
+          end
+          opts.on('--help', 'Display this command specific help output') do |help|
+            parsed['help'] = true
+          end
+          opts.on('--all', 'List signed certificates') do |a|
```

Comment:
Is this also going to list revoked certs like the old command?

Reply:
Yep, I just forgot about that. Should I change this to list all certificates, then? And then add an output line ``Revoked Certificates:`` ?